nemo/utils/config_utils.py

```diff
@@ -0,0 +1,86 @@
+# Copyright (c) 2020, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import copy
+from dataclasses import is_dataclass
+
+from omegaconf import DictConfig, OmegaConf, open_dict
+
+from nemo.core.config.modelPT import ModelPTConfig
+
+
+def update_model_config(model_cls: ModelPTConfig, update_cfg: DictConfig):
```

Comment:
[TODO] document this helper. Simply called by ModelPT to smartly merge the ModelPT dataclass with the "update" class. 

Question - should we provide a flag to avoid dropping the dataloaders / optim subconfigs if the "update_cfg" doesnt have those sections but the ModelPT dataclass has them ? Might be useful / harmful. Needs discussion.

Reply:
Yeah, generally I think it would be good if we could discuss this function and how should if work in all if/else cases...

Reply:
What is the difference between `update_model_dataclass` and `update_model_config`? Aren't those two covering the same functionality?

Reply:
I didn't want to clutter up ModelPT + I want to allow users to use this function without being reliant on ModelPT. So they are split and ModelPT simply forwards the call.